rules/msk_module_backend.go

```diff
@@ -0,0 +1,148 @@
+package rules
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/terraform-linters/tflint-plugin-sdk/hclext"
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+)
+
+// MskModuleBackendRule checks whether an MSK module has an S3 backend defined with a key that has as suffix the name of the team.
+type MskModuleBackendRule struct {
+	tflint.DefaultRule
+}
+
+// NewMskModuleBackendRule returns a new rule.
+func NewMskModuleBackendRule() *MskModuleBackendRule {
+	return &MskModuleBackendRule{}
+}
+
+// Name returns the rule name.
+func (r *MskModuleBackendRule) Name() string {
+	return "msk_module_backend"
+}
+
+// Enabled returns whether the rule is enabled by default.
+func (r *MskModuleBackendRule) Enabled() bool {
+	return true
+}
+
+// Severity returns the rule severity.
+func (r *MskModuleBackendRule) Severity() tflint.Severity {
+	return tflint.ERROR
+}
+
+// Link returns the rule reference link.
+func (r *MskModuleBackendRule) Link() string {
+	return ""
+}
+
+// Check checks whether.
+func (r *MskModuleBackendRule) Check(runner tflint.Runner) error {
+	path, err := runner.GetModulePath()
+	if err != nil {
+		return fmt.Errorf("getting module path: %w", err)
+	}
+	if !path.IsRoot() {
+		// This rule does not evaluate child modules.
+		return nil
+	}
+
+	// This rule is an example to get attributes of blocks other than resources.
+	content, err := runner.GetModuleContent(&hclext.BodySchema{
+		Blocks: []hclext.BlockSchema{
+			{
+				Type: "terraform",
+				Body: &hclext.BodySchema{
+					Blocks: []hclext.BlockSchema{
+						{
+							Type:       "backend",
+							LabelNames: []string{"type"},
+							Body: &hclext.BodySchema{
+								Attributes: []hclext.AttributeSchema{
+									{Name: "key"},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}, nil)
+	if err != nil {
+		return fmt.Errorf("getting module content: %w", err)
+	}
+
+	backend := findBackendDef(content)
+	if backend == nil {
+		err := runner.EmitIssue(r, "an s3 backend should be configured for a kafka MSK module", hcl.Range{})
+		if err != nil {
+			return fmt.Errorf("emitting issue: backend missing: %w", err)
+		}
```

Comment:
This check looks so odd to me (just a note on the SDK, nothing for us to do): there was an issue running the linter, but we can't show it to you. It would be nice if there was a helper like `EmitIssueOrAbort` to just end the runner, it's not like there's much we can do to recover :shrug: